### PR TITLE
Fix exceptions while generating projects and on run config validation

### PR DESCRIPTION
**Pull request by sdubov**
## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
- Fix exception on startup while gathering SAM CLI information; replace synchronous execution on EDT with modifying project model after project is initialized
- Fix validation for template.yaml file to be able to call SAM deploy action
- Fix sync calls on run configuration validation to use async calls
- Add tests to verify SamVersion evaluation and getFileInfo logic

## Motivation and Context
Async calls on startup and run configuration could freeze UI thread. Move all heavy logic to async tasks to make UI responsible.

## Related Issue(s)
None

## Testing
Environment:
```
Mac OS Mojave 10.14.5 (18F132)
```
- Check exceptions on creating new project from SAM Template
- Check on open existing templates
- Check on creating sub-modules with SAM project
- Check validation for creating run configuration and check for existing run configurations.

## Screenshots (if appropriate)
None.

## Checklist
- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 - **(Bug Fix)** Do not load proxy settings from Java system properties since it conflicts with IDE setting
 - **(Bug Fix)** Make sure we commit all open documents if using a file-based event input ([#910](https://github.com/aws/aws-toolkit-jetbrains/issues/910))
 - **(Bug Fix)** Fix being unable to open an empty credentials/config file for editing
+- **(Bug Fix)** Fix runtime exceptions on project startup and run configuration validation
```
Comment (abrooksv) [inline CHANGELOG.md]: Change log is auto-generated, run `./gradlew newBugfix --console plain`

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunConfiguration.kt [hunk 2]
```diff
@@ -58,7 +68,19 @@ class LocalLambdaRunConfiguration(project: Project, factory: ConfigurationFactor
...
+            ?: throw RuntimeConfigurationError(message("sam.cli_not_configured"))
+
+        val promise = SamVersionCache.evaluate(executablePath)
+
+        if (promise.isPending) {
+            logger.info { "Validation will proceed asynchronously for SAM CLI version" }
+            throw RuntimeConfigurationError(message("lambda.run_configuration.sam.validation.in_progress"))
+        }
+
+        val version = promise.blockingGet(0)!!
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunConfiguration.kt]: I assume this is null-able due to Java API, even though we have it defined as `Promise<SemVer>`?

File: resources/resources/software/aws/toolkits/resources/localized_messages.properties [hunk 3]
```diff
@@ -269,6 +271,8 @@ sam.init.template.dynamodb_cookiecutter.description=Sample SAM Template to inter
 sam.init.error.no.project.basepath=Unable to determine project basepath
 sam.init.error.subfolder_not_one=The SAM init root folder ''{0}'' should have only one sub folder.
 sam.init.error.no.virtual.file=Unable to resolve location
+sam.init.error.solution.create.fail=Unable to create solution
```
Comment (abrooksv) [inline resources/resources/software/aws/toolkits/resources/localized_messages.properties]: Rider message slipped in

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunConfiguration.kt (hunk 2 above)
Comment (sdubov) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunConfiguration.kt]: Correct. Here we guarantee that result is set over a check for `isPending()` above.

File: resources/resources/software/aws/toolkits/resources/localized_messages.properties (hunk 3 above)
Comment (sdubov) [reply resources/resources/software/aws/toolkits/resources/localized_messages.properties]: Removed.

**Comment by abrooksv** [comment]
Odd, the windows build got broken...but you didnt touch the fake SAM on Windows

```
  Expected: (an instance of java.lang.IllegalStateException and exception with message a string containing "SAM CLI did not provide any output") 
       but: an instance of java.lang.IllegalStateException <com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'ECHO': was expecting ('true', 'false' or 'null') 
   at [Source: (String)"ECHO is off. 
  "; line: 1][ column: 5]> is a com.fasterxml.jackson.core.JsonParseException 
  Stacktrace was: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'ECHO': was expecting ('true', 'false' or 'null') 
   at [Source: (String)"ECHO is off. 
  "; line: 1][ column: 5] 
    at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1804) 
    at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:679) 
    at 
```

**Comment by sdubov** [comment]
The fake SAM CLI tool returns output ECHO state when calling `echo ` if empty message is set. Fixed to return empty string instead.

**Comment by sdubov** [comment]
Please ignore the last commit review for now. It looks like it does not handle newline. I'll rework.

**Pull request merged**